File: 879/879.py
# Write a function that reverses a string. The input string is given as an array of characters s.

class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        b, e = 0, len(s) - 1
        while b <= e:
            temp = s[b]
            s[b] = s[e]
            s[e] = temp
            b += 1
            e -= 1

File: 879/879.rb
# Write a function that reverses a string. The input string is given as an array of characters s.

# @param {Character[]} s
# @return {Void} Do not return anything, modify s in-place instead.
def reverse_string(s)
    front = 0
    back = -1
    while front < s.size / 2
        temp = s[front]
        s[front] = s[back]
        s[back] = temp
        front += 1
        back -= 1
    end
end
